src/PublicApiGenerator/MethodNameBuilder.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.CodeDom;
+using System.Linq;
+using Mono.Cecil;
+using static System.String;
+
+namespace PublicApiGenerator
+{
+    public static class MethodNameBuilder
+    {
+        public static string AugmentMethodNameWithMethodModifierMarkerTemplate(MethodDefinition methodDefinition,
+            MemberAttributes attributes)
+        {
+            var name = CSharpOperatorKeyword.Get(methodDefinition.Name);
+
+            if (methodDefinition.DeclaringType.IsInterface)
+            {
+                return name;
+            }
+
+            bool? isNew = null;
+            var baseType = methodDefinition.DeclaringType.BaseType;
+            while (baseType != null)
+            {
+                var typeDef = baseType as TypeDefinition;
+                // too simple, improve
+                isNew = typeDef?.Methods.Any(e => e.Name.Equals(methodDefinition.Name, StringComparison.Ordinal) && e.Parameters.Count == methodDefinition.Parameters.Count);
+                if (isNew is true)
+                {
+                    break;
+                }
+                baseType = typeDef?.BaseType;
+            }
+
+            return (attributes & MemberAttributes.ScopeMask, isNew, methodDefinition.IsVirtual,
```

Comment:
it looks like it was copied from another file. Is it worth to extract in a method?

Reply:
I'm not a huge fan of generalizing too early. I'd wait and see how the property cases look like and then decide